File: Code/src/main/Java/Student.java
/**
 * Student is a class that consist of student information
 *
 */

import java.text.DecimalFormat;
import java.util.ArrayList;
import java.util.HashSet;

public class Student {

    private String major;
    private double majorGPA;
    private String totalGPA;
    private int numOfClasses;
    private ArrayList<Class> list;
    private HashSet<String> names;
    private HashSet<String> currentClasses;

    public Student(){
        major = "undeclared";
        majorGPA =0.0;
        totalGPA = "";
        numOfClasses = 0;
        list = new ArrayList();
        names = new HashSet<>();
        currentClasses = new HashSet<>();
    }

    /**
     * add class names
     */
    public void addNames(){
        for (Class one:list) {
            names.add(one.name);
        }
        names.add("NONE");
    }

    /**
     * @return class names
     */
    public  HashSet<String> getNames(){
        return names;
    }

    /**
     * @return major
     */
    public String getMajor() {
        return major;
    }

    /***
     * set major
     * @param major
     */
    public void setMajor(String major) {
        this.major = major;
    }

    /**
     * @return majorGPA
     */
    public String getMajorGPA() {
        DecimalFormat df2 = new DecimalFormat("#.##");
        return df2.format(majorGPA);
    }

    /**
     * set major
     * @param major
     */
    public void setMajorGPA(double major) { majorGPA = major; }

    /**
     * @return overall GPA
     */
    public String getTotalGPA() {
        return totalGPA;
    }

    /**
     * set overall GPA
     * @param totalGPA
     */
    public void setTotalGPA(String totalGPA) {
        this.totalGPA = totalGPA;
    }

    /***
     * @return number of classes
     */
    public int getNumOfClasses() {
        return numOfClasses;
    }

    /**
     * set number of classes
     * @param numOfClasses
     */
    public void setNumOfClasses(int numOfClasses) {
        this.numOfClasses = numOfClasses;
    }

    /**
     * @return class list
     */
    public ArrayList<Class> getList() {
        return list;
    }

    /**
     * @return list of current classes
     */
    public HashSet<String> getCurrentClasses(){return currentClasses;}

    public void setList(ArrayList<Class> list) {
        this.list = list;
    }
}

File: Code/js/Student.js
/**
 * Student.js is a class that consist of student information
 *
 */

class student {
    student(){
        this.major = "undeclared";
        this.majorGPA = 0.0;
        this.totalGPA = "";
        this.numOfClasses = 0;
        this.list = [];
        this.names = [];
        this.currentClasses = [];
    }


    getNames() {
        return this.names;
    }


    getMajor() {
        return this.major;
    }


    setMajor(major) {
        this.major = major;
    }

//TODO- gotta format output #.##

    getMajorGPA() {
        return this.majorGPA;
    }


    setMajorGPA(major) {
        this.majorGPA = major;
    }


    getTotalGPA() {
        return this.totalGPA;
    }


    setTotalGPA(totalGPA) {
        this.totalGPA = totalGPA;
    }


    getNumOfClasses() {
        return this.numOfClasses;
    }


    setNumOfClasses(numOfClasses) {
        this.numOfClasses = numOfClasses;
    }


    getList() {
        return this.list;
    }


    getCurrentClasses() {
        return this.currentClasses;
    }

    setCurrentClasses(currClasses) {
        this.currentClasses = currentClasses;
    }

//@param arrayList of taken classes

    setList(list) {
        this.list = list;
        //console.log('student.setList:'+this.list);
    }


    addNames() {
        var nlist = [];
        for(var element in this.list){
            nlist.push(this.list[element].name);
        }
        this.names = nlist;
    }
}

/**
 arraylist notes
 --------------------
 var array = [];
 array.push(val);
 array.pop(val)

 hashset/hashmap notes
 -------------------------
 var hash = {
  1:true,
  2:true,
  7:true
  //etc...
};

 var checkValue = function(value){
  return hash[value] === true;
};

 checkValue(7); // => true
 checkValue(3); // => false
 */